### PR TITLE
AWS: Improving getting existing VPC and subnet

### PR DESCRIPTION
**Pull request by gust1n**
Had some troubles creating a default AWS Kubernetes cluster into an existing VPC with multiple subnets. Current code will unfortunately fetch all of them which ofc fails. This solution works for me, I'm open to suggestions on how to improve it.

### CODE REVIEW

File: cluster/aws/util.sh [hunk 1]
```diff
@@ -45,7 +47,11 @@ function get_vpc_id {
 }
 
 function get_subnet_id {
-  python -c "import json,sys; lst = [str(subnet['SubnetId']) for subnet in json.load(sys.stdin)['Subnets'] if subnet['VpcId'] == '$1']; print ''.join(lst)"
+  python -c "import json,sys; lst = [str(subnet['SubnetId']) for subnet in json.load(sys.stdin)['Subnets'] if subnet['VpcId'] == '$1' and subnet['AvailabilityZone'] == '$2']; print ''.join(lst)"
```
Comment (brendandburns) [inline cluster/aws/util.sh]: fix indentation.

File: cluster/aws/util.sh [hunk 2]
```diff
@@ -45,7 +47,11 @@ function get_vpc_id {
...
+}
+
+function get_cidr {
+  python -c "import json,sys; lst = [str(subnet['CidrBlock']) for subnet in json.load(sys.stdin)['Subnets'] if subnet['VpcId'] == '$1' and subnet['AvailabilityZone'] == '$2']; print ''.join(lst)"
```
Comment (brendandburns) [inline cluster/aws/util.sh]: fix indentation.

File: cluster/aws/util.sh [hunk 3]
```diff
@@ -383,10 +389,15 @@ function kube-up {
...
   echo "Using VPC $VPC_ID"
 
-  SUBNET_ID=$($AWS_CMD describe-subnets | get_subnet_id $VPC_ID)
+  SUBNET_ID=$($AWS_CMD describe-subnets | get_subnet_id $VPC_ID $ZONE)
   if [[ -z "$SUBNET_ID" ]]; then
 	  echo "Creating subnet."
-	  SUBNET_ID=$($AWS_CMD create-subnet --cidr-block 172.20.0.0/24 --vpc-id $VPC_ID --availability-zone ${ZONE} | json_val '["Subnet"]["SubnetId"]')
+	  SUBNET_ID=$($AWS_CMD create-subnet --cidr-block $INTERNAL_IP_BASE.0/24 --vpc-id $VPC_ID --availability-zone ${ZONE} | json_val '["Subnet"]["SubnetId"]')
+  else
+    EXISTING_CIDR=$($AWS_CMD describe-subnets | get_cidr $VPC_ID $ZONE)
```
Comment (brendandburns) [inline cluster/aws/util.sh]: fix indentation.

**Comment by brendandburns** [comment]
Looks reasonable to me.  Please fix the indentation in the places mentioned so that it is consistent.  Then I'm happy to merge.

Thanks
--brendan

File: cluster/aws/util.sh (hunk 1 above)
Comment (gust1n) [inline cluster/aws/util.sh]: Seems correct to me?

File: cluster/aws/util.sh (hunk 2 above)
Comment (gust1n) [inline cluster/aws/util.sh]: Here too

**Comment by gust1n** [comment]
Rebased and squashed, PTAL. Also maybe @justinsb since he is the AWS wiz.

**Comment by justinsb** [comment]
LGTM (by way of background, $ZONE was initially actually a region; now this is fixed we can probably add more filtering)

I've also seen this (pre-existing) scariness, although that'll be a fix for another day:
```
--private-ip-address 172.20.0.1${i}
```

**Comment by gust1n** [comment]
ping

File: cluster/aws/util.sh (hunk 2 above)
Comment (brendandburns) [inline cluster/aws/util.sh]: as above.

Comment (brendandburns) [inline cluster/aws/util.sh]: oops, nm, it's the one on 37 that's the outlier...

**Comment by brendandburns** [comment]
Merged.

**Pull request merged**